File: 1161/main.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        self.levels = list[int]()
        self.dfsSum(root, 0)

        maxLevelValue = root.val
        maxLevelIndex = 0

        for level in range(len(self.levels)):
            if maxLevelValue < self.levels[level]:
                maxLevelValue = self.levels[level]
                maxLevelIndex = level

        return maxLevelIndex + 1

    def dfsSum(self, node: Optional[TreeNode], level: int) -> None:
        if not node:
            return

        if level < len(self.levels):
            self.levels[level] += node.val
        else:
            self.levels.append(node.val)

        self.dfsSum(node.left, level + 1)
        self.dfsSum(node.right, level + 1)

File: 1161/main.ts
function maxLevelSum(root: TreeNode | null): number {
	const levels: number[] = [];
	const dfsSum = (node: TreeNode | null, level: number): void => {
		if (node === null) return;

		if (level in levels) {
			levels[level] += node.val;
		} else {
			levels[level] = node.val;
		}

		dfsSum(node.left, level + 1);
		dfsSum(node.right, level + 1);
	};

	dfsSum(root, 0);

	let maxLevelIndex = 0;
	let maxLevelValue = levels[0];

	for (let level = 1; level < levels.length; level++) {
		if (maxLevelValue < levels[level]) {
			maxLevelValue = levels[level];
			maxLevelIndex = level;
		}
	}

	return maxLevelIndex + 1;
}

/*   *   *   *   *   *   *   *   *   *   */
/*   *   *   *   *   *   *   *   *   *   */

class TreeNode {
	val: number;
	left: TreeNode | null;
	right: TreeNode | null;
	constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
		this.val = val === undefined ? 0 : val;
		this.left = left === undefined ? null : left;
		this.right = right === undefined ? null : right;
	}
}
